tests/ert_tests/data/test_measured_data.py

```diff
@@ -110,42 +110,98 @@ def test_remove_failed_realizations(
+        pytest.param(
+            pd.DataFrame(data=[[1, None, 3], [4, 5, 6]], index=["OBS", "STD"]),
+            pd.DataFrame(data=[[7.0, 8.0, 9.0]], index=[1]),
+            pd.DataFrame(
+                data=[[1.0, 3.0], [4.0, 6.0], [7.0, 9.0]],
+                index=["OBS", "STD", 1],
+                columns=[0, 2],
+            ),
+            id="none-in-OBS-row",
+        ),
+        pytest.param(
+            pd.DataFrame(data=[[1, 2, 3], [4, None, 6]], index=["OBS", "STD"]),
+            pd.DataFrame(data=[[7.0, 8.0, 9.0]], index=[1]),
             pd.DataFrame(
                 data=[[1.0, 3.0], [4.0, 6.0], [7.0, 9.0]],
                 index=["OBS", "STD", 1],
                 columns=[0, 2],
             ),
+            id="none-in-STD-row",
+        ),
+        pytest.param(
+            pd.DataFrame(data=[[1, 2, 3], [4, 5, 6]], index=["OBS", "STD"]),
+            pd.DataFrame(data=[[7.0, np.nan, 9.0]], index=[1]),
+            pd.DataFrame(
+                data=[[1.0, 3.0], [4.0, 6.0], [7.0, 9.0]],
+                index=["OBS", "STD", 1],
+                columns=[0, 2],
+            ),
+            id="nan-in-measured-data",
+        ),
+        pytest.param(
+            pd.DataFrame(data=[[1, 2, 3], [4, 5, 6]], index=["OBS", "STD"]),
+            pd.DataFrame(data=[[7.0, np.inf, 9.0]], index=[1]),
+            pd.DataFrame(
+                data=[[1.0, 3.0], [4.0, 6.0], [7.0, 9.0]],
+                index=["OBS", "STD", 1],
+                columns=[0, 2],
+            ),
+            id="inf-in-measured-data",
+        ),
+        pytest.param(
+            pd.DataFrame(data=[[1, 2, 3], [4, 5, 6]], index=["OBS", "STD"]),
+            pd.DataFrame(data=[[7.0, -np.inf, 9.0]], index=[1]),
+            pd.DataFrame(
+                data=[[1.0, 3.0], [4.0, 6.0], [7.0, 9.0]],
+                index=["OBS", "STD", 1],
+                columns=[0, 2],
+            ),
+            id="negative-inf-in-measured-data",
+        ),
+        pytest.param(  # This should raise ValueError.
+            pd.DataFrame(data=[[1, 2, 3], [4, 5, 6]], index=["OBS", "STD"]),
+            pd.DataFrame(data=[[np.inf, np.inf, np.inf]], index=[1]),
+            pd.DataFrame(data=[], index=["OBS", "STD", 1]),
+            id="all-inf-in-measured-data"
         ),
     ],
 )
 def test_remove_inactive_observations(
     input_dataframe,
+    data,
     expected_result,
     monkeypatch,
     facade,
     measured_data_setup,
-    valid_data,
 ):
     input_dataframe.columns = _set_multiindex(input_dataframe)
-    measured_data_setup(valid_data, input_dataframe, monkeypatch)
+    measured_data_setup(data, input_dataframe, monkeypatch)
     md = MeasuredData(facade, ["obs_key"])
 
     expected_result.columns = _set_multiindex(expected_result)
     expected_result = pd.concat({"obs_key": expected_result}, axis=1)
 
-    md.remove_inactive_observations()
-    assert md.data.equals(expected_result)
+    try:
```

Comment:
Perhaps we should test when measured data contains all infs / nans / None's?

This does not look pretty, so probably not the best way to test for this.
You know of a better way?